src/App.js

```diff
@@ -1,16 +1,85 @@
-import React from 'react';
+import React, {useState} from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
+import ColorChoice from './components/ColorChoice';
 
 const App = () => {
+  const[entriesData, setEntriesData] = useState(chatMessages);
+
+  const likeEntry = (id) => {
+    setEntriesData((prev)=>{
+      return prev.map((entry)=>{
+        if(id === entry.id){
+          return{
+            ...entry,
+            liked: !entry.liked
+          };}
+        else{
+          return entry;
+        }
+        });
+    });
+  };
+
+  const totalLikes= entriesData.reduce((total, entry)=>{
+    if(entry.liked === true){
+      total +=1;
+    }
+    return total;
+  }, 0);
+
+  const [senderColors, setSenderColors] = useState({
+    [chatMessages[0].sender]: 'black',
+    [chatMessages[1].sender]: 'black'
+  });
```

Comment:
👍 Nice introduction of a piece of state to track the current colors. Note that this code assumes there are at least two messages in the chat already, and that they are sent by the two different people in the conversation. What would happen at the start of the conversation before any messages are sent? What if one sender sends multiple messages at the start?